src/chat.rs

```diff
@@ -2145,6 +2147,69 @@ pub async fn marknoticed_chat(context: &Context, chat_id: ChatId) -> Result<()>
     Ok(())
 }
 
+pub(crate) async fn mark_old_messages_as_noticed(
```

Comment:
Documentation comment is needed. "Marks messages preceding outgoing messages as noticed" or something. Also tell what the `msgs` argument is.
Otherwise it is completely unclear why it does not simply accept `Context` and does some magic to fix unnoticed messages that should be noticed without reading the code.